compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/AMDAIELinalgFunctionOutlining.cpp

```diff
@@ -0,0 +1,115 @@
+// Copyright 2024 The IREE Authors
+//
+// Licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include "iree-amd-aie/IR/AMDAIEOps.h"
+#include "iree-amd-aie/Transforms/AMDAIEUtils.h"
+#include "iree-amd-aie/Transforms/Passes.h"
+#include "mlir/Conversion/FuncToLLVM/ConvertFuncToLLVM.h"
+#include "mlir/Conversion/FuncToLLVM/ConvertFuncToLLVMPass.h"
+#include "mlir/Dialect/Linalg/IR/LinalgInterfaces.h"
+#include "mlir/Dialect/Linalg/Transforms/Hoisting.h"
+#include "mlir/Dialect/Linalg/Transforms/Transforms.h"
+#include "mlir/Dialect/Vector/IR/VectorOps.h"
+
+#define DEBUG_TYPE "iree-amdaie-linalg-function-outlining"
+
+namespace mlir::iree_compiler::AMDAIE {
+
+namespace {
+
+/// Utility to outline the linalg compute op.
+static FailureOr<func::FuncOp> outlinedToAFunction(
+    IRRewriter &rewriter, ModuleOp &moduleOp, linalg::LinalgOp &computeOp) {
+  // Form outlined FuncName.
+  std::string computeName = "";
+  if (isMatmul(computeOp)) {
+    computeName = "_matmul";
+  } else if (isElementwise(computeOp)) {
+    computeName = "_elementwise";
+  } else {
+    return failure();
```

Comment:
I've removed `failure()` altogether and am now simply skipping those compute ops from the `walk` before invoking this function to outline. Let me know if that sounds okay to you.

Reply:
It's ok. I think it'd still be good to improve the possible situation in N months time when someone is trying to understand why matmul transpose is running out of PM, whereas matmul is not -- if rather than silently not outlining matmul transpose, you made the pass fail for unconsidered ops, the person might be able to avoid the situation. 

I guess it's a trade off between failing too often when you shouldn't and letting issues silently slip through.

Reply:
Makes sense.

Instead of failing I'm emitting a remark.

Failing this pass shouldn't be an apt way - because outlining is supposed to be an improvement in the performance (time/space) and shouldn't really prevent the pipeline to move further in compilation.

Reply:
I don't like remarks, I think I'd prefer complete silence than a remark, because there are ops for which we don't want outlining (fill). Matmul and matmul transpose should be handled in the same way here, eventually.

Reply:
Sure, I've removed the remark and going ahead with skipping the unsupported ops for now. Please take a look.